.github/workflows/ci-main.yml

```diff
@@ -90,10 +90,15 @@ jobs:
                 scons "$@" build="$CFG" ctool="$CC" cxxtool="$CXX" cxx_std="$CXX_STD" \
                   extra_flags_config="-pipe" forum_user_handler=true \
                   nls=false enable_lto="$LTO" system_lua="$SYS_LUA" force_color=true \
-                  jobs=2 --debug=time glibcxx_debug=true glibcxx_assertions=true
+                  jobs=2 --debug=time glibcxx_debug=true glibcxx_assertions=true compile_db=true
               }
               build strict=true wesnoth boost_unit_tests
 
+              build cdb
+              # disable all warnings since we have coverage on them in a standard build, and clang-tidy
+              # triggers false positive compiler warnings that clang itself won't
+              run-clang-tidy -use-color -j 2 -extra-arg="-w" '^(?!.*src\/modules|.*build\/).*$'
```

Comment:
Presumably the backslashes do nothing. Trailing `.*$` does not look useful either.

Anything wrong with `^(?!src/modules/|build/)`?

Reply:
@soliton- The compilation database uses absolute paths, so the minimum I can reduce it to is `'^(?!.*src/modules/|build/)'`

Reply:
'^(?!.*src/modules/|.*build/)', actually.

Reply:
How do you get the compilation DB to contain absolute file paths? I didn't manage to.

Oh I see run-clang-tidy puts directory and file together before applying the regex. Makes no sense to me but oh well.